### PR TITLE
dts: drop HAS_DTS

### PR DESCRIPTION
**Pull request by gmarull**
HAS_DTS has become a redundant option. All Zephyr architectures now select this option, and no out-of-tree architectures are supported, meaning devicetree has become a de-facto requirement. There are actually many Zephyr areas that are not guarded, and expect devicetree to always be available regardless of HAS_DTS.

Depends on https://github.com/zephyrproject-rtos/zephyr/pull/63799

### CODE REVIEW

**Review by carlocaione — DISMISSED** [review]

**Review by tejlmand — DISMISSED** [review]
afaik there are still users not having DTS downstream.

Would like review from @mped-oticon / @daor-oti first.

**Comment by gmarull** [comment]
> afaik there are still users not having DTS downstream.
> 
> Would like review from @mped-oticon / @daor-oti first.

Considering that all architectures require DTS, and that we do not support oot archs, I suspect this must be a heavily patched Zephyr. What this PR shows is that Zephyr as is now, can live without this option because it's doing nothing in practice.

**Review by fabiobaltieri — DISMISSED** [review]
Hard to justify keeping this in if nothing upstream builds without it. Worth running it through the arch meeting I guess but it feels like a reasonable requirement.

**Review by daor-oti — CHANGES_REQUESTED** [review]
As @aescolar writes, out of tree architectures are supported and used. We would not be able to use Zephyr without this.
We also cannot support DTS and therefor the HAS_DTS configure option is important to us, so please do not remove that.

**Comment by fabiobaltieri** [comment]
> We also cannot support DTS and therefor the HAS_DTS configure option is important to us [...]

Could you add more context about it? What's the blocker?

**Comment by gmarull** [comment]
> > and that we do not support oot archs
> 
> We support out of tree architectures :)

In a module, like e.g. in example-application? I've heard from @carlescufi many times that this is indeed not supported.

**Comment by gmarull** [comment]
> As @aescolar writes, out of tree architectures are supported and used. We would not be able to use Zephyr without this. We also cannot support DTS and therefor the HAS_DTS configure option is important to us, so please do not remove that.

Could you provide more details on your usage? I'm surprised you're able to build something useful without devicetree nowadays, it is implicitly required in many Zephyr areas.

Btw, I realized this PR will not break anything anyway (maybe serial backend kconfig option for shell?). Can you please try this patch?

**Comment by gmarull** [comment]
> > We also cannot support DTS and therefor the HAS_DTS configure option is important to us [...]
> 
> Could you add more context about it? What's the blocker?

+1, Devicetree can't be "not supported", it's either you want to pick it or not :smile:

**Comment by carlescufi** [comment]
> > > and that we do not support oot archs
> > 
> > 
> > We support out of tree architectures :)
> 
> In a module, like e.g. in example-application? I've heard from @carlescufi many times that this is indeed not supported.

That depends on how you define "architecture". There is no `ARCH_ROOT` to be able to plug-in external architectures without modifying the in-tree code. So I assume @daor-oti and @mped-oticon are referring to modified trees that include an extra architecture?

**Review by carlescufi — DISMISSED** [review]
As @fabiobaltieri already stated, Devicetree is mandatory in Zephyr today already, we just haven't said it openly. I am in favor of moving forward with this change, and if there are out-of-tree architectures that prefer to avoid it then I am sure we can accomodate changes that make their lives easier, but not an explicit "disable DT" switch.

**Comment by carlescufi** [comment]
> As @aescolar writes, out of tree architectures are supported and used. We would not be able to use Zephyr without this. We also cannot support DTS and therefor the HAS_DTS configure option is important to us, so please do not remove that.

@daor-oti We do not want to break out-of-tree users, but at this point can you please explain why you need `HAS_DTS` in particular? This is barely used and clearly does not cover any specific use-cases anymore due to the fact that its in-tree use is partial at best.

**Comment by gmarull** [comment]
If there's any part of the system that should be disabled, a specific Kconfig option can be created (e.g. to autodeclare devices?). The HAS_DTS option is currently to indicate DT is supported, but there's no companion option to turn DT off (like it happens with other HAS_*). Also, all platforms upstream require DT now, meaning that a non-DT route is never tested nor even considered when adding new features (e.g. pinctrl makes DT mandatory, but it does not depend on it). As mentioned earlier, if you can provide some details, I'm pretty confident this can move forward one way or the other.

**Review by bjarki-andreasen — DISMISSED** [review]
I believe this is the right path for Zephyr. We should deprecate and remove options and features to maintain stability and scalability as they loose prevalence in the project.

**Comment by daor-oti** [comment]
> > We also cannot support DTS and therefor the HAS_DTS configure option is important to us [...]
> 
> Could you add more context about it? What's the blocker?

@fabiobaltieri Our compiler and our build system is not able to support DTS, as there are some incompatibilities with the macros used in Device Tree. Reworking and or fixing these issues is a non-trivial task, and thus will require significant work on our part, if at all possible.

**Comment by daor-oti** [comment]
> > As @aescolar writes, out of tree architectures are supported and used. We would not be able to use Zephyr without this. We also cannot support DTS and therefor the HAS_DTS configure option is important to us, so please do not remove that.
> 
> Could you provide more details on your usage? I'm surprised you're able to build something useful without devicetree nowadays, it is implicitly required in many Zephyr areas.

@gmarull We are using only a limited part of Zephyr, and not many device drivers from Zephyr either, which is why we are able to build something useful. But we still relying on being able to disable DTS for those few drivers that we do use from Zephyr. As we a medical class device, we are not patching Zephyr, but using the out-of-tree options for the parts that are unique to out platform, and which would be of not use to anyone else, as they are completely custom to our platform.

**Comment by daor-oti** [comment]
> > As @aescolar writes, out of tree architectures are supported and used. We would not be able to use Zephyr without this. We also cannot support DTS and therefor the HAS_DTS configure option is important to us, so please do not remove that.
> 
> @daor-oti We do not want to break out-of-tree users, but at this point can you please explain why you need `HAS_DTS` in particular? This is barely used and clearly does not cover any specific use-cases anymore due to the fact that its in-tree use is partial at best.

@carlescufi My two previous comments explains why we cannot use DTS at this time. Let me know if you need more explanation/information.

>
> So I assume @daor-oti and @mped-oticon are referring to modified trees that include an extra architecture?

@carlescufi Not entirely sure what you mean by 'modified trees', but we do not patch nor modify Zephyr in our use case, we rely on the out-of-tree features for selecting our arch, board and soc, together with our custom drivers, all of which is custom and of no use to anyone else using Zephyr.

**Comment by aescolar** [comment]
We need to be a bit careful with this type of things. Zephyr is used by quite a lot of people in another way than the plain vanilla upstream repository using only Zephyr provided infrastructure and components and just adding an app.
It is true that practically all testing we do upstream is for "the default recommended way of using it", but we do support lots of other ways.
* We support extending Zephyr with out of tree *everything* (libraries, drivers, boards, socs, archs, other compiler and linker integrations..).
* Zephyr provides a lot of optional things, and most users pick and chose a very small subset. It is still possible to only use the kernel.

My humble opinion is that keeping to support this is both beneficial and necessary for the project:

* We need to remember that Zephyr does not target a simple and uniform ecosystem with 1 or 2 of "each", but one where custom solutions are more the norm. Trying to support everything in tree, or force all bells and whistles in the out of tree versions is impossible/counterproductive.
* Quite a few areas in Zephyr are not mature enough for what people building final products need, so they are forced to implement their own replacements. And unfortunately they need to do that in a short time-frame to get to market (i.e. off-tree and avoiding all unnecessary flair they can)
* Some of these out of tree improvements/add-ons eventually become part of Zephyr. And even upstream we are  adding support for new things all the time. Having the possibility to disable some parts of Zephyr is a great help as a way to start prototyping/developing in a simplified environment.
* Keeping some parts optional also forces us to keep the architecture cleaner, with those optional components more modularized.

-----

For this particular HAS_DTS issue, it seems the upstream benefit of removing it is very minor.  While the drawback for some users seems to be much greater.
So I would slow down a bit with this clean up. An option may be to put it in the roadmap, but I would certainly not hurry it.

**Comment by gmarull** [comment]
I think that to block this change we need a better understanding of what users do. Upstream doesn’t need to carry all downstream technical debts. For example, are we saying that we have users with a non ISO C compiler? Because AFAIK Devicetree macros do not make use of any non-C standard feature. util.h does much more macro black magic, for example. 

Also, what is the exact problem with this patch if applied? This has not been answered yet.

**Comment by fabiobaltieri** [comment]
> @fabiobaltieri Our compiler and our build system is not able to support DTS, as there are some incompatibilities with the macros used in Device Tree. Reworking and or fixing these issues is a non-trivial task, and thus will require significant work on our part, if at all possible.

@daor-oti could you provide more details of what precisely is the problem? (and how/if it breaks this patch specifically) The project scope is very large and long term sustainability is a big issue, we already "support" multiple compilers ([eight](https://docs.zephyrproject.org/latest/develop/toolchains/index.html)?) but only test with ~two. We certainly can't rely on reviewers to manually spot potentially breaking changes, it'd be useful to have more details about your use case and see if it can be supported in a sustainable way.

**Review by finikorg — DISMISSED** [review]

**Comment by gmarull** [comment]
> We need to be a bit careful with this type of things. Zephyr is used by quite a lot of people in another way than the plain vanilla upstream repository using only Zephyr provided infrastructure and components and just adding an app. It is true that practically all testing we do upstream is for "the default recommended way of using it", but we do support lots of other ways.
> 
> * We support extending Zephyr with out of tree _everything_ (libraries, drivers, boards, socs, archs, other compiler and linker integrations..).
> * Zephyr provides a lot of optional things, and most users pick and chose a very small subset. It is still possible to only use the kernel.

Devicetree is currently not optional, in the sense that we don't have a `CONFIG_DTS that can be set to `n` (unlike e.g. `CONFIG_OF` in Linux). This patch is removing an option that in practice does very few things, it basically prevents some auto-declarations. These auto-declarations should, in fact, be empty if you have 0 devicetree nodes. Note that some CMake scripts (dts.cmake) do allow skipping certain DT parts, like board.dts files, but this PR does not change those parts.

> 
> My humble opinion is that keeping to support this is both beneficial and necessary for the project:
> 
> * We need to remember that Zephyr does not target a simple and uniform ecosystem with 1 or 2 of "each", but one where custom solutions are more the norm. Trying to support everything in tree, or force all bells and whistles in the out of tree versions is impossible/counterproductive.
> * Quite a few areas in Zephyr are not mature enough for what people building final products need, so they are forced to implement their own replacements. And unfortunately they need to do that in a short time-frame to get to market (i.e. off-tree and avoiding all unnecessary flair they can)
> * Some of these out of tree improvements/add-ons eventually become part of Zephyr. And even upstream we are  adding support for new things all the time. Having the possibility to disable some parts of Zephyr is a great help as a way to start prototyping/developing in a simplified environment.
> * Keeping some parts optional also forces us to keep the architecture cleaner, with those optional components more modularized.
> 
> For this particular HAS_DTS issue, it seems the upstream benefit of removing it is very minor. While the drawback for some users seems to be much greater. So I would slow down a bit with this clean up. An option may be to put it in the roadmap, but I would certainly not hurry it.

I think we just need a bit more information. I feel that this patch will not remove the possibility of not using devicetree because this Kconfig option does very few things nowadays. There are many other headers using the DT APIs without any kind of guarding.

**Comment by daor-oti** [comment]
@gmarull and @fabiobaltieri As I have written previously, it fails to build with errors in expansion of the DEVICE_DT_NAME_GET macro, and potentially others after this. As the compiler is custom, no more information is available and would not be of much use anyway. This is not something which is trivial to fix for us, and thus removing the HAS_DTS config will render Zephyr unusable for us, at least until a fix/workaround can be found. Therefor we would prefer this not to be merged, until at suitable replacement solution can be found, and we are currently investigating what we can do to resolve our issue.

**Comment by gmarull** [comment]
> @gmarull and @fabiobaltieri As I have written previously, it fails to build with errors in expansion of the DEVICE_DT_NAME_GET macro, and potentially others after this. As the compiler is custom, no more information is available and would not be of much use anyway. This is not something which is trivial to fix for us, and thus removing the HAS_DTS config will render Zephyr unusable for us, at least until a fix/workaround can be found. Therefor we would prefer this not to be merged, until at suitable replacement solution can be found, and we are currently investigating what we can do to resolve our issue.

Compiler logs would help (AFAIK logs are not proprietary). Without further information, there's not much we can do. Also, if you say you're not using Devicetree, I wonder why you are encountering issues with a DT-based macro. Maybe it should never have been called in the first place, so we're blocking a valid change, perhaps because HAS_DT is papering over a real bug. It may not be trivial for you to fix, but potentially we can help to resolve the problem with enough information.

**Comment by daor-oti** [comment]
@gmarull Could you elaborate as to why this is so urgent? Does the HAS_DTS config break anything for anyone if left as is for now?

**Comment by gmarull** [comment]
> @gmarull Could you elaborate as to why this is so urgent? Does the HAS_DTS config break anything for anyone if left as is for now?

Whether it's urgent or not doesn't matter. The problem here is that we're blocking valid changes for unknown reasons, and problems like this seem to be an increasing trend, see https://github.com/zephyrproject-rtos/zephyr/issues/63738. As you can imagine, it's not sustainable to block changes like this in an open-source project. I'm happy to help, propose alternatives, etc. But, I don't even have a single log. This started as "our compiler cannot support devicetree" to later we have a problem with "DEVICE_DT_NAME_GET", there's clearly information missing here.

**Comment by nordicjm** [comment]
I would argue that this would count as a stable API change and would need approval as such - I'm not talking about the actual changes of this PR itself, and would suggest that if it is agreed to continue with this, it should be deprecated for 2 releases then removed after that time.

**Comment by henrikbrixandersen** [comment]
> The problem here is that we're blocking valid changes for unknown reasons

Well, removing existing functionality that is only used downstream doesn't necessarily classify the change as "valid". This is exactly why we have reviews of proposed changes. Breaking out-of-tree/downstream usage of Zephyr should be avoided if possible. 

I'd say we could use more details on the exact nature of the compilation failures mentioned here, but I do believe the raised objections are sufficient for putting this change on hold until a solution is found.

**Comment by gmarull** [comment]
> > The problem here is that we're blocking valid changes for unknown reasons
> 
> Well, removing existing functionality that is only used downstream doesn't necessarily classify the change as "valid". This is exactly why we have reviews of proposed changes. Breaking out-of-tree/downstream usage of Zephyr should be avoided if possible.
> 

I think this is yet another case of pushing the stability boundary too far. `HAS_DTS` is not an oot feature per-se, its implications are not even documented. The reality is that it seems to be an obsolete feature that right now has the only effect of preventing auto-declarations of some DT devices. What surprises me is that despite users claiming _they don't use devicetree_, these auto-declarations still expand to something (meaning devicetree is actually plugged in), see how it works:

```c
#define Z_MAYBE_DEVICE_DECLARE_INTERNAL(node_id)                               \                                                                                     
         extern const struct device DEVICE_DT_NAME_GET(node_id);                                                                                                      

DT_FOREACH_STATUS_OKAY_NODE(Z_MAYBE_DEVICE_DECLARE_INTERNAL) 
```

So `DT_FOREACH_STATUS_OKAY_NODE` will do nothing on a system without devicetree nodes.

> I'd say we could use more details on the exact nature of the compilation failures mentioned here, but I do believe the raised objections are sufficient for putting this change on hold until a solution is found.

To find a solution, we need details, and there's basically nothing right now (or information that makes me think the problem is something else).

**Comment by gmarull** [comment]
> I would argue that this would count as a stable API change and would need approval as such - I'm not talking about the actual changes of this PR itself, and would suggest that if it is agreed to continue with this, it should be deprecated for 2 releases then removed after that time.

I'm fine with deprecation, this is not changing any APIs though.

**Comment by fabiobaltieri** [comment]
> As I have written previously, it fails to build with errors in expansion of the DEVICE_DT_NAME_GET macro, and potentially others after this. As the compiler is custom, no more information is available and would not be of much use anyway.

That's a very convenient way of blocking any feature anywhere at any time. :-) Imagine if we had a dozen users with custom requirements that they cannot explain... Could you describe the technical limitation in any way? Sure we could come up with a half decent workaround.

**Comment by daor-oti** [comment]
As we have now gotten a little time to investigate this further, I now have some more information.

The essential issue is regarding these lines and macros from device.h:
`#define Z_MAYBE_DEVICE_DECLARE_INTERNAL(node_id)                               \
	extern const struct device DEVICE_DT_NAME_GET(node_id);

DT_FOREACH_STATUS_OKAY_NODE(Z_MAYBE_DEVICE_DECLARE_INTERNAL)`

The essence of the issue is, that the devicetree_generated.h header is empty, and thus the DT_FOREACH_OKAY_HELPER macro does not exist, hence compile failures as can be seen from the following log extract:

> In file included from rf/apps/sample_app/src/app_cmds_mfi_disturbance.c:11:
In file included from rf/zephyr/include/zephyr/drivers/entropy.h:26: rf/zephyr/include/zephyr/device.h:1000:1: error: type specifier missing, defaults to 'int'; ISO C99 and later do not support implicit int [-Werror,-Wimplicit-int] DT_FOREACH_STATUS_OKAY_NODE(Z_MAYBE_DEVICE_DECLARE_INTERNAL)
^
int
rf/zephyr/include/zephyr/devicetree.h:2498:41: note: expanded from macro 'DT_FOREACH_STATUS_OKAY_NODE'
#define DT_FOREACH_STATUS_OKAY_NODE(fn) DT_FOREACH_OKAY_HELPER(fn)

The reason for why the devicetree_generated.h is empty, and a method that will fit into our build environment is currently being investigated.

As soon as I have more information to provide, I will let you know.

**Comment by gmarull** [comment]
@daor-oti thanks for the investigation, that starts to make more sense. I guess the responsible for this problem is:

https://github.com/zephyrproject-rtos/zephyr/blob/4d598683972a3601c2e90b53d58ffbb6e33e99ea/cmake/modules/dts.cmake#L141-L144

However, DT API is not guarded nor offers stubs. Could you quickly try to add an empty board_name.dts file in your board directory, e.g. with

```
/dts-v1/;

/ {

};
```

And see if you observe the same problem? Could you share the devicetree_generated.h in such scenario?

Thanks

**Comment by tejlmand** [comment]
> I guess the responsible for this problem is:

if the problem is indeed related to an empty file and can be improved by creating a dts skeleton with no devices, then I would say we should update `dts.cmake` to do so, instead of a purely empty file.

**Comment by gmarull** [comment]
> > I guess the responsible for this problem is:
> 
> if the problem is indeed related to an empty file and can be improved by creating a dts skeleton with no devices, then I would say we should update `dts.cmake` to do so, instead of a purely empty file.

agreed. the empty file could actually be

```
/dts-v1/;

/ {

};
```

However, I've found that this fix is needed to make this work https://github.com/zephyrproject-rtos/zephyr/pull/63799

**Comment by gmarull** [comment]
@daor-oti please try latest PR revision

File: cmake/modules/dts.cmake
Comment (tejlmand) [inline cmake/modules/dts.cmake:133]: Is this message needed ?

After all the `stub.dts` will also be printed here:
https://github.com/zephyrproject-rtos/zephyr/blob/8c6a3dbe37abd9b905ac89a375b97a811470dcbd/cmake/modules/dts.cmake#L182

as:
```
"-- Found BOARD.dts: <zephyr-base>/misc/stub.dts
```

so should be visible enough that a dts stub file is being used.

File: misc/stub.dts
Comment (tejlmand) [inline misc/stub.dts:1]: as this file is being used (and printed) as being a board dts file, should we then place with the boards folder instead, like here: `boards/board-stub.dts`.

https://github.com/zephyrproject-rtos/zephyr/tree/main/boards

**Review by tejlmand — COMMENTED** [review]
Thanks for looking into a solution on this.
Some minor proposals for making it cleaner.

File: cmake/modules/dts.cmake
Comment (fabiobaltieri) [reply cmake/modules/dts.cmake:133]: I think you want to comment in https://github.com/zephyrproject-rtos/zephyr/pull/63799

File: misc/stub.dts
Comment (tejlmand) [reply misc/stub.dts:1]: That would result in following message during CMake configure:
```
"-- Found BOARD.dts: <zephyr-base>/boards/board-stub.dts
```

**Comment by daor-oti** [comment]
> @daor-oti please try latest PR revision

@gmarull Checking it now...

**Review by daor-oti — CHANGES_REQUESTED** [review]
@gmarull These latest changes looks good, and we now get a populated devicetree_generated.h header, and as such we are able to build without issues. Thank you for providing a resolution for this and for your patience.

@gmarull Could you remove the part in the commit message that reads 'and no out-of-tree architectures are supported'? because they are supported. Also, if possible, add a statement that an empty devicetree is supported?

File: misc/stub.dts
Comment (gmarull) [reply misc/stub.dts]: done

File: cmake/modules/dts.cmake
Comment (gmarull) [reply cmake/modules/dts.cmake]: done

**Review by tejlmand — DISMISSED** [review]

**Review by daor-oti — APPROVED** [review]
Thanks again, looks good.

**Review by aescolar — DISMISSED** [review]

**Review by henrikbrixandersen — DISMISSED** [review]

**Review by carlescufi — DISMISSED** [review]

**Review by fabiobaltieri — DISMISSED** [review]

File: boards/board-stub.dts
Comment (mbolivar-ampere) [inline boards/board-stub.dts:1]: Move to dts/common and include skeleton.dtsi

**Review by mbolivar-ampere — CHANGES_REQUESTED** [review]
LGTM, just one issue

File: boards/board-stub.dts
Comment (gmarull) [reply boards/board-stub.dts:1]: @tejlmand agree?

Comment (mbolivar-ampere) [reply boards/board-stub.dts:1]: The main requirement from my side is to get skeleton.dtsi in there. The location is negotiable

Comment (gmarull) [reply boards/board-stub.dts:1]: > The main requirement from my side is to get skeleton.dtsi in there. The location is negotiable

Can you please elaborate why skeleton.dtsi is a requirement for this particular case? I just want to understand the reasons before making the change.

Comment (mbolivar-ampere) [reply boards/board-stub.dts:1]: 1. I want to keep consistency
2. An empty root node is disallowed by the DT spec -- we can be flexible with missing things like `model` and `compatible`, but I at least want to have `#address-cells` and `#size-cells` defined with sensible defaults. See Table 3.2 Root Node Properties in the DT spec

**Review by daor-oti — CHANGES_REQUESTED** [review]
Please know that I am very much for this change and the PR, but unfortunately we have hit another snag, as we are now not getting our drivers linked in, when building with this PR included. This relates to my previous mentioning of 'our toolchain does not support device tree'.

Would there be any chance that we could postpone this PR and change, until we have had more time to dig deeper into the issues we are having, and how we can solve our toolchain issues? Sorry for the inconvenience.

**Comment by fabiobaltieri** [comment]
@gmarull, @daor-oti would it be feasible to set up an isolated test case to reproduce your setup? That way we could carry on with these cleanups without worring too much about breaking your downstream configuration.

**Comment by daor-oti** [comment]
@fabiobaltieri That would be an ideal solution, unfortunately we have a custom toolchain consisting if in-house developed tools, which we cannot share. Unless I am misunderstanding what you are suggesting? We are though looking into the issues and trying to see if we can come up with a solution that will allow us to not block this PR for too long, or possibly suggest a workaround.

**Comment by fabiobaltieri** [comment]
> @fabiobaltieri That would be an ideal solution, unfortunately we have a custom toolchain consisting if in-house developed tools, which we cannot share. Unless I am misunderstanding what you are suggesting?

Not really, I understand that some tooling is custom but stuff like the `Z_DEVICE_DT_DEV_ID` breakage could certainly be reproduced upstream with a custom board in a test or something similar. I'm sure you could not cover everything, but you could certainly cover something.

> We are though looking into the issues and trying to see if we can come up with a solution that will allow us to not block this PR for too long, or possibly suggest a workaround.

Sure, it's just that if we can have an upstream test that reproduces the specific condition in isolation we could have some confidence that it won't break again.

Also, is there a linker error you could paste here?

**Comment by daor-oti** [comment]
There are no linker errors and the target seems to be running, but is apparently missing the device drivers. We are investigating and will update here as soon as we have anything useful. Sorry for the holdup and the delays.

File: boards/board-stub.dts
Comment (tejlmand) [reply boards/board-stub.dts:1]: > @tejlmand agree?

From dts viewpoint, then `dts/common` looks better, but from users point of view where the `board-stub.dts` functions as a board's dts file when a board is missing a dts file, then I believe the boards folder seems most appropriate.

Just like regular boards has their dts file as `boards/<arch>/<board>/<boardname>.dts`, then the stub is in the board hierarchy.

I do not have a strong opinion and can accept both locations, but based on above reasoning then I have a preference for the `boards` location.

Comment (mbolivar-ampere) [reply boards/board-stub.dts:1]: OK @tejlmand -- how about boards/common/board-stub.dts? Putting this right in boards/ feels like showing everyone who wants to look at zephyr's boards something that 99.999% of them will not care about.

Comment (henrikbrixandersen) [reply boards/board-stub.dts:1]: Perhaps just `boards/common/stub.dts`?

Comment (gmarull) [reply boards/board-stub.dts]: followed @henrikbrixandersen suggestion, and skeleton.dtsi is included.

**Comment by gmarull** [comment]
> There are no linker errors and the target seems to be running, but is apparently missing the device drivers. We are investigating and will update here as soon as we have anything useful. Sorry for the holdup and the delays.

Please provide any logs or information, so it can be investigated. It looks like a bug that was being papered by HAS_DT, same as the one that this PR fixes.

**Review by fabiobaltieri — APPROVED** [review]

**Review by daor-oti — APPROVED** [review]
Thanks once again for your patience. 
We seem to have resolved our issues and all should be good from our side.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Review by bjarki-andreasen — APPROVED** [review]

**Review by mbolivar-ampere — APPROVED** [review]

**Pull request merged**
